src/lib.rs

```diff
@@ -718,12 +714,12 @@ pub struct Function {
 #[cfg_attr(feature = "serialize", derive(Serialize))]
 #[cfg_attr(feature = "deserialize", derive(Deserialize))]
 pub struct EntryPoint {
-    /// The stage in the programmable pipeline this entry point is for.
-    pub stage: ShaderStage,
-    /// Name identifying this entry point.
-    pub name: String,
-    /// The function to be used.
-    pub function: Handle<Function>,
+    /// Early depth test for fragment stages.
+    pub early_depth_test: Option<EarlyDepthTest>,
+    /// Workgroup size for compute stages
+    pub workgroup_size: [u32; 3],
+    /// The function body.
```

Comment:
Minor, but maybe crucial for people new to the codebase: this isn't actually only the function body, but also the signature and all other details associated with a function. I think this comment is more confusing than having no comment.